exercises/a-safe-space-for-nulls/problem/problem.md

```diff
@@ -0,0 +1,77 @@
+
+You will have a variable named `$user` available in your PHP script. We will put it there, each time your program runs  and it will be populated with random data each time it runs.
+
+Sometimes, a number of the properties may not exist and will be null. 
+
+With the null safe operator it is possible to access variables like so:
+
+```php
+$capitalPopulation = $country?->capital?->population;
+```
+
+If the `capital` property is null, the variable `$capitalPopulation` will also contain null. Previously, without the null safe operator this would be achieved like so:
+
+```php
+$capitalPopulation = null;
+if ($city->capital !== null) {
+    $capitalPopulation = $country->capital->population;
+}
+```
+
+The `User` class, for which the `$user` variable holds an instance of, has the following signature:
+
+```php
+class User
+{
+    public string $firstName;
+    public string $lastName;
+    public ?int $age = null;
+    public ?Address $address = null;
+}
+
+class Address
+{
+    public int $number;
+    public string $addressLine1;
+    public ?string $addressLine2 = null;
+}
+```
+
+Note also the `Address` class which the property `$user->address` may be an instance of, or it may be null.
+
+Export the `$user` data to a CSV with the following columns:
+
+`First Name`, `Last Name`, `Age`, `House num`, `Addr 1`, `Addr 2`
+
+ * The CSV should be comma delimited
+ * The columns should read exactly as above, any mistake will trigger a failure
+ * There should be one row for the column headers and one for the data
+ * Any properties which are null on the user should be printed as empty fields in the CSV
+ * The file should be named `users.csv` and exist next to your submission file (eg in the same directory)
+ 
+And finally, the most important part, all properties which may be `NULL` should be accessed using the null safe operator!
+
+### Advantages of the null safe operator
+
+* Much less code for simple operations where null is a valid value
+* If the operator is part of a chain anything to the right of the null will not be executed, the statements will be short-circuited.
+* Can be used on methods where null coalescing cannot `$user->getCreatedAt()->format() ?? null` where `getCreatedAt()` could return null or a `\DateTime` instance
+
+----------------------------------------------------------------------
+## HINTS
+
+Remember your program will be passed no arguments. There will be a `User` object populated for you under the variable `$user`.
+It is available at the beginning of your script.
+
+Documentation on the Null Safe Operator can be found by pointing your browser here:
+[https://www.php.net/manual/en/language.oop5.basic.php#language.oop5.basic.nullsafe]()
+
+----------------------------------------------------------------------
+## EXTRA
+
+We have not given any hints regarding writing to a CSV file, as we are not testing you on that. You can do it however you please.
```

Comment:
>You can do it however you please.

I'm fine with this, as in I understand it, but it did come to mind whether this would be understandable to someone whose first language isn't English, or whether it translates well. 

Happy to take your lead on it tbh I have far less experience with languages and cultures as you 😄

Reply:
Will have a think!